packages/core/src/components/html-select/htmlSelect.tsx

```diff
@@ -60,8 +60,9 @@ export class HTMLSelect extends React.PureComponent<IHTMLSelectProps> {
         );
 
         const optionChildren = options.map(option => {
-            const { value, label }: IOptionProps = typeof option === "object" ? option : { value: option };
-            return <option key={value} value={value} children={label || value} />;
+            const { className: optionClassName, value, label }: IOptionProps =
```

Comment:
Had to mix up the variable name to avoid shadowing the `className` variable on [L59](https://github.com/palantir/blueprint/pull/2783/files#diff-a8ae96697a4c86c0af1cafd4a6894e7cR59).